src/lib/__tests__/InstantSearch-test.js

```diff
@@ -537,6 +547,37 @@ describe('dispose', () => {
 
     expect(search.helper).not.toBe(null);
   });
+
+  it('should remove the listeners on the helper', done => {
+    const search = new InstantSearch({
+      indexName: '',
+      searchClient: {
+        search() {
+          return Promise.resolve({
+            results: [
+              {
+                query: 'fake query',
+              },
+            ],
+          });
+        },
+      },
+    });
+
+    search.on('error', () => {
+      done.fail('InstantSearch should not throw an error.');
+    });
+
+    search.start();
+
+    expect(search.helper).not.toBe(null);
```

Comment:
This assertion is redundant with the previous test, it is not related to this test. We can remove it.